akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -52,7 +51,7 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
    */
   def tryPullSubStreams(): Unit
 
-  private var streamStates = new immutable.TreeMap[Int, StreamState]
+  private val streamStates = new java.util.Hashtable[Int, StreamState](settings.maxConcurrentStreams)
```

Comment:
This will box the Int key. Would it be better to use `scala.collection.mutable.LonMap`?

Reply:
I can try that.

Reply:
It's hard to measure a difference but LongMap seems slightly faster, indeed.